pkg/cmd/pulumi/state_setignorechanges.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024, Pulumi Corporation.
```

Comment:
Introducing this new command seems somewhat orthogonal to the rest of the PR.  I'd love to see that pulled out into a separate PR so we can iterate on the UX for that and think some more about what exactly we want that to look like without having to block this PR on that.

Reply:
I can pull it out but realistically this PR is likely blocked since without that command, it's hard to manipulate `IgnoreChanges` for use in a refresh. On the other point about `pulumi state set ...`, I did discuss this with Luke since I agree it's likely the right call. However, it feels like a bigger piece of work and we already have `pulumi state unprotect`, which is a special case in its own right, so I was hoping we could find a way to tackle that later (and either deprecate or alias this command, for instance). Thoughts?

Reply:
After more discussion I'm going to pull this out and we can discuss separately :+1: